### ISSUE TITLE
Unable to fine-tune using train.py

### ISSUE THREAD

**Issue by marchollinger** [gotutiyan/gec-t5#1]
I am trying to fine-tune the model `gotutiyan/gec-t5-base-clang8` on my own dataset using `train.py`.

From my understanding, it should be possible to pass `gotutiyan/gec-t5-base-clang8` to  the `--model-id` parameter of the training script. 

The script does run when I do this, however, the loss is reported as `nan` and the learning rate as `0`. The loss is reported correctly during validation but not during training.

Using another GEC checkpoint like `anujraymajhi/t5-GEC-6` works. The learning rate and loss are reported just like when training from `google-t5/t5-base`.

Should it be possible to use `train.py` to fine-tune based on your checkpoints? If it should, what could be the reason for this problem?

**Comment by gotutiyan** [comment]
Thank you for reporting the issue!

Yes, it is correct to pass the model ID to `--model-id`.

I noticed that although I performed some refactoring about a month ago, I forgot to update `train.py`.
The dataset loading part was broken, which might be related to the issue you encountered. (However, it's strange that the learning rate is 0, as that seems unrelated to this issue regarding dataset loading.)

This issue has now been fixed. At least in my environment, the loss does not become NaN.

Could you please try running it again? Please let me know if the problem remains.

**Comment by marchollinger** [comment]
Thank you for the quick reply!

After your change, the `nan` issue still happens for me when using the default parameters. Meaning the model_id `google/t5-v1_1-base` does not work for me currently. 

The loss is now also `nan` during validation. During training, the loss sometimes shows a number for a split second but then goes back to `nan`. 

Using `t5-base` which was the default before your change still works as expected. 

~Using `gotutiyan/gec-t5-base-clang8` still doesn't work but `gotutiyan/gec-t5-small-clang8` does.~

I don't think it has anything to do with my data because I also tried it with a very small mock dataset which lead to the same issue.

**Comment by marchollinger** [comment]
I realized that `gotutiyan/gec-t5-small-clang8` does not work on larger amounts of data. 

When I try to use my real data, the loss stays `nan` most of the time but for some epochs, a final loss is reported even though it showed `nan` until the end of the epoch. 

The only case where there wasn't a single `nan` in the output was for a mock dataset with only five sentences.

**Comment by marchollinger** [comment]
I have solved the issue. According to [this issue](https://discuss.huggingface.co/t/t5-fp16-issue-is-fixed/3139) the `nan` happens when using t5 with `fp16`.

Changing `fp16` to `bf16` in my accelerate config solved it for me.
